public/javascripts/common/SpeedLimit.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * An indicator that displays the speed limit of the current position's nearest road.
+ *
+ * @param {StreetViewPanorama} panorama Panorama object
+ * @param {function} coords Function that returns current longitude and latitude coordinates
+ * @param {function} isOnboarding Function that returns a boolean on whether the current mission is the tutorial/onboarding task
+ * @returns {SpeedLimit} SpeedLimit object with updateSpeedLimit function, container, speedLimit object with number and sub (units, e.g. 'mph'),
+ * speedLimitVisible boolean
+ */
+function SpeedLimit(panorama, coords, isOnboarding) {
+    let self = this;
+
+    function _init() {
+        self.container = document.getElementById("speed-limit-sign");
+        self.speedLimit = {
+            number: "",
+            sub: ""
+        };
+        self.speedLimitVisible = false
+        updateSpeedLimit();
+
+        // Listen for position changes.
+        panorama.addListener('position_changed', positionChange);
+    }
+
+    function updateSpeedLimit() {
+        self.container.querySelector("#speed-limit").innerText = self.speedLimit.number;
+        self.container.querySelector("#speed-limit-sub").innerText = self.speedLimit.sub;
+        self.container.style.display = self.speedLimitVisible ? "flex" : "none";
+    }
+
+    function findClosestRoad(data, lat, lon) {
+        // Filter to only be roads, and not foot paths/walk ways.
+        const roadHighwayTypes = [
```

Comment:
Let's define this at the top of the file instead of inside a function. Kinda makes sense to put constants at the top, and it makes it easier to read what the function is doing.